crates/core/src/push.rs

```diff
@@ -144,7 +144,9 @@ pub fn zip_file(
 ///
 /// The `.git` folders are always skipped.
 pub fn filter_ignored_files(root_directory_path: impl AsRef<Path>) -> Vec<PathBuf> {
-    let files_to_copy = Arc::new(Mutex::new(Vec::with_capacity(100)));
+    let (tx, rx) = std::sync::mpsc::channel::<PathBuf>();
```

Comment:
```suggestion
    use std::sync::mpsc;

    let (tx, rx) = mpsc::channel::<PathBuf>();
```

maybe import the mpsc module to reduce visual clutter a bit?